Favor/Favor/Sources/Scenes/Auth/ViewControllers/SignUpVC.swift

```diff
@@ -0,0 +1,196 @@
+    return textField
+  }()
+  
+  private lazy var pwTextField: FavorTextField = {
+    let textField = FavorTextField()
+    textField.delegate = self
+    textField.placeholder = "비밀번호"
+    textField.updateMessage("영문, 숫자 혼용 8자 이상", for: .info)
+    textField.keyboardType = .asciiCapable
+    textField.autocapitalizationType = .none
+    return textField
+  }()
+  
+  private lazy var pwValidateTextField: FavorTextField = {
+    let textField = FavorTextField()
+    textField.delegate = self
+    textField.placeholder = "비밀번호 확인"
+    textField.keyboardType = .asciiCapable
+    textField.autocapitalizationType = .none
+    return textField
+  }()
+  
+  private lazy var nextButton: LargeFavorButton = {
+    let button = LargeFavorButton(with: .white, title: "다음")
+    button.isEnabled = false
+    return button
+  }()
+  
+  private lazy var vStack: UIStackView = {
+    let stackView = UIStackView()
+    stackView.axis = .vertical
+    stackView.spacing = 72.0
+    [
+      self.emailTextField,
+      self.pwTextField,
+      self.pwValidateTextField,
+      self.nextButton
+    ].forEach {
+      stackView.addArrangedSubview($0)
+    }
+    return stackView
+  }()
+  
+  // MARK: - Life Cycle
+  
+  // MARK: - Binding
+  
+  func bind(reactor: SignUpReactor) {
+    // Action
+    self.emailTextField.rx.text
+      .orEmpty
+      .distinctUntilChanged()
+      .map { Reactor.Action.emailTextFieldUpdate($0) }
+      .bind(to: reactor.action)
+      .disposed(by: self.disposeBag)
+    
+    self.pwTextField.rx.text
+      .orEmpty
+      .distinctUntilChanged()
+      .map { Reactor.Action.passwordTextFieldUpdate($0) }
+      .bind(to: reactor.action)
+      .disposed(by: self.disposeBag)
+    
+    self.pwValidateTextField.rx.text
+      .orEmpty
+      .distinctUntilChanged()
+      .map { Reactor.Action.checkPasswordTextFieldUpdate($0) }
+      .bind(to: reactor.action)
+      .disposed(by: self.disposeBag)
+    
+    self.nextButton.rx.tap
+      .map { Reactor.Action.nextButtonTap }
+      .bind(to: reactor.action)
+      .disposed(by: self.disposeBag)
+    
+    // State
+    reactor.state.asObservable()
+      .map { $0.isEmailValid }
+      .distinctUntilChanged()
+      .skip(1)
+      .subscribe(onNext: { emailValidate in
+        switch emailValidate {
+        case .empty, .invalid:
+          self.emailTextField.updateMessage(emailValidate.description, for: .error)
+        case .valid:
+          self.emailTextField.updateMessage(nil, for: .info)
+        }
+      })
+      .disposed(by: self.disposeBag)
+    
+    reactor.state.asObservable()
+      .map { $0.isPasswordValid }
+      .distinctUntilChanged()
+      .skip(1)
+      .subscribe(onNext: { passwordValidate in
+        switch passwordValidate {
+        case .empty, .invalid:
+          self.pwTextField.updateMessage(passwordValidate.description, for: .error)
+        case .valid:
+          self.pwTextField.updateMessage(nil, for: .info)
+        }
+      })
```

Comment:
이건 개인적인 생각인데,
`subscribe`보다는 `bind`를 사용하는게 어떨까?
```suggestion
      .bind { owner, passwordValidate in
        switch passwordValidate {
        case .empty, .invalid:
          owner.pwTextField.updateMessage(passwordValidate.description, for: .error)
        case .valid:
          owner.pwTextField.updateMessage(nil, for: .info)
        }
      }
```

driver를 쓰기에는 계속 asDriver로 바꾸는 번거로움이 있을 것 같아서,
어차피 VC에서 하는 작업들은 거의 `MainThread`에서 작업할테니까..
UI Binding을 할때에는 어차피 `Error`도 방출하지 않고?

의견 부탁해!

Reply:
음.. 이왕 바꿀거면 driver로 쓰는 게 좋지 않을까?
`bind`는 메인 쓰레드에서 동작한다는 보장이 없는걸로 알고 있어서!

Reply:
```suggestion
      .map { $0.isPasswordIdentical }
      .asDriver(onErrorJustReturn: .identical)
      .distinctUntilChanged()
      .skip(1)
      .drive(onNext: { [weak self] isPasswordIdentical in
        switch isPasswordIdentical {
        case .empty, .different:
          self?.pwValidateTextField.updateMessage(isPasswordIdentical.description, for: .error)
        case .identical:
          self?.pwValidateTextField.updateMessage(nil, for: .info)
        }
      })
      .disposed(by: self.disposeBag)
```

Reply:
오케이! UI관련 작업은 `drive`를 통해서 구독하는걸로 하자 
그 외의 작업들은 `subscribe`나 `bind`를 써서 적절히 스레드를 옮기거나 해줘도 될 것 같아